e2e/kube.go

```diff
@@ -76,32 +76,29 @@ func getClusterKubeconfigBytes(ctx context.Context, resourceGroupName, clusterNa
 }
 
 // this is a bit ugly, but we don't want to execute this piece concurrently with other tests
-func ensureDebugDaemonset(ctx context.Context, kube *Kubeclient) error {
-	manifest := getDebugDaemonset()
-	var ds v1.DaemonSet
-
-	if err := yaml.Unmarshal([]byte(manifest), &ds); err != nil {
-		return fmt.Errorf("failed to unmarshal debug daemonset manifest: %w", err)
+func ensureDebugDaemonsets(ctx context.Context, kube *Kubeclient) error {
+	manifests := getDebugDaemonsetManifests()
+	for _, manifest := range manifests {
+		if err := createDebugDeployment(ctx, kube, manifest); err != nil {
+			return err
+		}
 	}
 
-	desired := ds.DeepCopy()
-	_, err := controllerutil.CreateOrUpdate(ctx, kube.Dynamic, &ds, func() error {
-		ds = *desired
-		return nil
-	})
+	return nil
+}
 
-	if err != nil {
-		return fmt.Errorf("failed to apply debug daemonset: %w", err)
+func getDebugDaemonsetManifests() []string {
+	return []string{
+		getDebugDaemonsetTemplate(hostNetworkDebugPodNamePrefix, "nodepool1", true),
+		getDebugDaemonsetTemplate(nonHostNetworkDebugPodNamePrefix, "nodepool2", false),
 	}
-
-	return nil
 }
 
-func getDebugDaemonset() string {
-	return `apiVersion: apps/v1
+func getDebugDaemonsetTemplate(deploymentName, targetNodeLabel string, isHostNetwork bool) string {
+	return fmt.Sprintf(`apiVersion: apps/v1
 kind: Deployment
```

Comment:
lol, can we change the naming if it's not actually a daemonset

Reply:
probably better for it to actually be a daemonset and then just only select for the system pool, that way we get new replicas if we get new system pool nodes

Reply:
ugh, I don't wanna make you do that here...

I guess if we want to switch it over to actual daemonset then we can leave the naming how it currently is

Reply:
Naming is inconsistent here, will fix